torch/nn/modules/linear.py

```diff
@@ -115,4 +116,99 @@ def __repr__(self):
             + ', in2_features=' + str(self.in2_features) \
             + ', out_features=' + str(self.out_features) + ')'
 
+
+class NoisyLinear(Module):
+    """Applies a noisy linear transformation to the incoming data:
+        :math:`y = (mu_w + sigma_w \cdot epsilon_w)x
+            + mu_b + sigma_b \cdot epsilon_b`
+    More details can be found in the paper `Noisy Networks for Exploration` _ .
+    Args:
+        in_features: size of each input sample
+        out_features: size of each output sample
+        bias: If set to False, the layer will not learn an additive bias.
+            Default: True
+        factorised: whether or not to use factorised noise.
+            Default: True
+        std_init: initialization constant for standard deviation component of
+            weights. If None, defaults to 0.017 for independent and 0.4 for
+            factorised. Default: None
+    Shape:
+        - Input: :math:`(N, in\_features)`
+        - Output: :math:`(N, out\_features)`
+    Attributes:
+        weight: the learnable weights of the module of shape
+            (out_features x in_features)
+        bias:   the learnable bias of the module of shape (out_features)
+    Examples::
+        >>> m = nn.NoisyLinear(20, 30)
+        >>> input = autograd.Variable(torch.randn(128, 20))
+        >>> output = m(input)
+        >>> print(output.size())
+    """
+    def __init__(self, in_features, out_features, bias=True, factorised=True, std_init=None):
+        super(NoisyLinear, self).__init__()
+        self.in_features = in_features
+        self.out_features = out_features
+        self.factorised = factorised
+        self.weight_mu = Parameter(torch.Tensor(out_features, in_features))
+        self.weight_sigma = Parameter(torch.Tensor(out_features, in_features))
+        if bias:
+            self.bias_mu = Parameter(torch.Tensor(out_features))
+            self.bias_sigma = Parameter(torch.Tensor(out_features))
+        else:
+            self.register_parameter('bias', None)
+        if not std_init:
+            if self.factorised:
+                self.std_init = 0.4
+            else:
+                self.std_init = 0.017
+        else:
+            self.std_init = std_init
+        self.reset_parameters(bias)
```

Comment:
Could you add `self.bias = bias` above and therefore run `self.reset_parameters()` without any arguments (same as `nn.Linear`)?